src/modules/pathfindingAlgorithms/dfs.ts

```diff
@@ -0,0 +1,54 @@
+// Implementation of Depth First Search algorithm
+
+import type { TileCords } from "@/types/CommonTypes";
+import { areStartAndGoalPlaced, getNeighbors, getStartAndGoalCords } from "@/modules/commonFunctions/searchingHelpers";
+
+/**
+ * Function implements Depth First Search algorithm.
+ *
+ * @param grid grid on which to perform algorithm,
+ * @param ingoreCells algorithm will ignore cells of given type,
+ * @returns All visited cellls with start and goal included
+ */
+export const DFSAlgorithm = (grid: string[][], ingoreCells: string[]) => {
+  if (!areStartAndGoalPlaced(grid)) {
```

Comment:
To juz jest zrobione w store wiec nie mamy mozliwosci juz bazowo odpalic symulacji jesli brakuje start albo goal